samples/bluetooth/central_uart/src/main.c

```diff
@@ -486,6 +480,46 @@ static int nus_client_init(void)
 BT_SCAN_CB_INIT(scan_cb, scan_filter_match, NULL,
 		scan_connecting_error, scan_connecting);
 
+static void try_to_connect_to_bonded_device(const struct bt_bond_info *info, void *user_data)
+{
+	LOG_INF("Bond found");
+
+	struct bt_le_conn_param *conn_params = BT_LE_CONN_PARAM_DEFAULT;
+	char addr[BT_ADDR_LE_STR_LEN];
+	size_t *count = user_data;
+
+	bt_addr_le_to_str(&info->addr, addr, sizeof(addr));
+
+	(*count)++;
+	int err = bt_conn_le_create(&info->addr, BT_CONN_LE_CREATE_CONN,
```

Comment:
I am not sure if using `bt_conn_le_create` directly here is the best approach. Consider configuring proper scan filters and starting scan in similar way as it's done here: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/ble_scan.c#L365

That would allow us to keep a single BLE scan execution path (simpler implementation, easier to maintain). With that approach we may even enable the new feature unconditionally (should not break anything)

Reply:
You could also consider supporting multiple BLE connections (e.g. with an early return if connection count is already equal to `CONFIG_BT_MAX_CONN`). Ref: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/ble_scan.c#L204

Reply:
You may need to update the `bt_scan` subsystem configuration too: https://github.com/nrfconnect/sdk-nrf/blob/main/applications/nrf_desktop/src/modules/Kconfig.ble_scan#L17

Reply:
@MarekPieta Thanks for your suggestions. I have added the address of the device to the scan filters.